lib/app/models/module.rb

```diff
@@ -2,139 +2,31 @@
-      end
-    end
-
-    class TestGroup
-      attr_accessor :test_set
-      attr_accessor :name
-      attr_accessor :overview
-      attr_accessor :input_instructions
-      attr_accessor :lock_variables
-      attr_accessor :id
-      attr_accessor :test_cases
-      attr_accessor :run_all
-      attr_accessor :run_skipped
-
-      def initialize(test_set, name, overview, input_instructions, lock_variables, run_all, run_skipped)
-        @test_set = test_set
-        @name = name
-        @id = name.gsub(/[^0-9a-z]/i, '')
-        @overview = overview
-        @run_all = run_all
-        @test_cases = []
-        @test_case_names = {}
-        @input_instructions = input_instructions
-        @lock_variables = lock_variables || []
-        @run_skipped = run_skipped
-      end
-
-      def add_test_case(sequence_name, parameters = {})
-        current_name = "#{test_set.id}_#{@id}_#{sequence_name}"
-        index = 1
-        while @test_case_names.key?(current_name)
-          index += 1
-          current_name = "#{test_set.id}_#{@id}_#{sequence_name}_#{index}"
-          raise 'Too many test cases using the same scenario' if index > 99
-        end
-
-        @test_case_names[current_name] = true
-
-        sequence = Inferno::Sequence::SequenceBase.descendants.find { |seq| seq.sequence_name == sequence_name }
-
-        raise "No such sequence: #{sequence_name}" if sequence.nil?
-
-        new_test_case = TestCase.new(current_name, self, sequence, parameters)
-
-        @test_cases << new_test_case
-
-        new_test_case
-      end
-    end
-
-    class TestCase
-      attr_accessor :id
-      attr_accessor :test_group
-      attr_accessor :sequence
-      attr_accessor :parameters
-
-      def initialize(id, test_group, sequence, parameters)
-        @id = id
-        @sequence = sequence
-        @test_group = test_group
-        @parameters = parameters
-      end
-
-      def title
-        if !@parameters[:title].nil?
-          @parameters[:title]
-        else
-          sequence.title
-        end
-      end
-
-      def description
-        if !@parameters[:description].nil?
-          @parameters[:description]
-        else
-          sequence.description
-        end
-      end
-
-      def variable_defaults
-        @parameters[:variable_defaults]
-      end
-    end
-
-    attr_accessor :name
-    attr_accessor :title
-    attr_accessor :hide_optional
-    attr_accessor :description
     attr_accessor :default_test_set
+    attr_accessor :description
     attr_accessor :fhir_version
+    attr_accessor :hide_optional
+    attr_accessor :name
+    attr_accessor :resources
     attr_accessor :test_sets
+    attr_accessor :title
 
-    def initialize(name, description, default_test_set, fhir_version, title)
-      @name = name
-      @description = description
-      @default_test_set = default_test_set
-      @fhir_version = fhir_version
+    def initialize(**params)
```

Comment:
Why  switch out explicit parameters for an map of parameters?

Reply:
In the existing code, the call site was performing logic to combine two parameters into one, and also contained some rather complex logic to perform additional initialization of the module. I wanted to move all that logic into the constructor, which would have meant increasing the parameter count from five to seven. Five parameters is already pretty bad, but seven is really bad. The parameters were already in a hash at the call site, so it seemed undesirable to pull them out and put them in a particular order rather than just passing the hash.

Reply:
Makes sense!